Add more info about missing plugins to migration guide (#3710)
openbao/openbao: website/content/docs/guides/migration.mdx

If your plugin isn't included, you may need to resort to some [workarounds](https://github.com/openbao/openbao/issues/257#issuecomment-3228461262).
This guide assumes no external plugins (external to Bao) are used.

:::

Take inventory before you stop Vault:

```
$ vault secrets list
$ vault auth list
```

Any mount whose type OpenBao does not ship is skipped when OpenBao reads the mount
table. This is easy to miss, because startup logs an error, a warning and

> `successfully mounted` for the same path:

Reviewer: ```suggestion
Any mount whose backing plugin OpenBao does not include as a built-in is mounted
but not initialized (skipped) at startup, leaving a stubbed backend that does
not serve any paths. Server logs will reflect this:
```

```
[ERROR] core: failed to create mount entry: path=aws/ error="plugin not found in the catalog: aws"
[WARN]  core: skipping plugin-based mount entry: path=aws/
[INFO]  core: successfully mounted: type=aws version="v2.0.3+builtin.vault" path=aws/
```

> Afterwards the path is still listed by `bao secrets list`, but reading it returns
> `No value found`, the same answer an unused path gives. Clean it up with
> `bao secrets disable <path>` or `bao auth disable <path>` once you have taken care of
> whatever it held.

Reviewer: ```suggestion
The skipped mount will remain listed under `bao secrets list` or `bao auth
list` but any request to it will return `No value found` as no paths have been
registered.

If no longer needed, remove the mount via `bao secrets disable <path>` or `bao
auth disable <path>`. Alternatively, if available, install the matching plugin
from [openbao-plugins](https://github.com/openbao/openbao-plugins) and run
`bao secrets tune <path>` or `bao auth tune <path>` using the `-plugin-version`
flag to point the mount at the correct plugin, then run `bao plugin reload
-mounts=<path>`.
```

Author: > but any request to it will return `No value found`

is not exact, I tried this out:

```sh
$ bao read aws/config/root
No value found at aws/config/root
$ bao list aws/roles
No value found at aws/roles
$ bao write aws/config/root access_key=dummy secret_key=dummy region=eu-central-1
Error writing data to aws/config/root: Error making API request.
```

`bao secrets|auth tune <path>` requires a `plugin register` and I tried it out:

```sh
$ bao plugin register -sha256=641ae1858c0f660b4c3761286d627e1dcfd33dc35e7c83eb81ca9050b5bdc6d8 -command=openbao-plugin-secrets-aws_linux_amd64_v1 -version=v0.3.1 secret aws
Success! Registered plugin: aws
$ bao secrets tune -plugin-version=v0.3.1 aws
Success! Tuned the secrets engine at: aws/
$ bao plugin reload -mounts=aws
Success! Reloaded mounts: [aws]
$ bao write aws/config/root access_key=dummy secret_key=dummy region=eu-central-1
Error writing data to aws/config/root: Error making API request.
```

but after restart 

```sh
$ docker restart lab-bao && bao operator unseal <shamir>
```

```sh
$ bao write aws/config/root access_key=dummy secret_key=dummy region=eu-central-1
Success! Data written to: aws/config/root
```

I think this behavior should be mentioned in the documentation or a separate issue should be opened for it.

Reviewer 2: @trutzonline Do you see any logs between:

```
$ bao secrets tune -plugin-version=v0.3.1 aws
Success! Tuned the secrets engine at: aws/
$ bao plugin reload -mounts=aws
Success! Reloaded mounts: [aws]
$ bao write aws/config/root access_key=dummy secret_key=dummy region=eu-central-1
Error writing data to aws/config/root: Error making API request.
```

that you can share, perhaps in trace mode?

Author: Here the complete trace log, I think the problem is `2026-08-07T15:46:25.198Z [TRACE] secrets.aws.aws_ed580048.aws: handle request: transport=gRPC path=config/root status=finished err="cannot write to storage during setup" took=1.460691ms` ...

```sh
2026-08-07T15:46:21.076Z [INFO]  proxy environment: http_proxy="" https_proxy="" no_proxy=""
2026-08-07T15:46:21.078Z [DEBUG] storage.raft.fsm: time to open database: elapsed=1.87511ms path=/openbao/data/vault.db
2026-08-07T15:46:21.085Z [DEBUG] core: set config: sanitized config="{\"allow_audit_log_prefixing\":false,\"api_addr\":\"http://127.0.0.1:8200\",\"cache_size\":0,\"cluster_addr\":\"http://127.0.0.1:8201\",\"cluster_cipher_suites\":\"\",\"cluster_name\":\"\",\"default_lease_ttl\":0,\"default_max_request_duration\":0,\"detect_deadlocks\":\"\",\"disable_cache\":false,\"disable_clustering\":false,\"disable_indexing\":false,\"disable_performance_standby\":false,\"disable_printable_check\":false,\"disable_sealwrap\":false,\"disable_sentinel_trace\":false,\"enable_response_header_hostname\":false,\"enable_response_header_raft_node_id\":false,\"enable_ui\":false,\"imprecise_lease_role_tracking\":false,\"introspection_endpoint\":false,\"listeners\":[{\"config\":{\"address\":\"0.0.0.0:8200\",\"tls_disable\":true},\"type\":\"tcp\"}],\"log_format\":\"\",\"log_level\":\"trace\",\"log_requests_level\":\"\",\"max_lease_ttl\":0,\"pid_file\":\"\",\"plugin_directory\":\"/openbao/plugins\",\"plugin_file_permissions\":0,\"plugin_file_uid\":0,\"raw_storage_endpoint\":true,\"seals\":[{\"disabled\":false,\"type\":\"shamir\"}],\"storage\":{\"cluster_addr\":\"http://127.0.0.1:8201\",\"disable_clustering\":false,\"raft\":{\"max_entry_size\":\"\"},\"redirect_addr\":\"http://127.0.0.1:8200\",\"type\":\"raft\"},\"unsafe_allow_api_audit_creation\":false,\"unsafe_cross_namespace_identity\":false}"
2026-08-07T15:46:21.086Z [DEBUG] storage.cache: creating LRU cache: size=0
2026-08-07T15:46:21.086Z [INFO]  core: Initializing version history cache for core
2026-08-07T15:46:21.089Z [DEBUG] cluster listener addresses synthesized: cluster_addresses=[0.0.0.0:8201]

==> OpenBao server configuration:

             Api Address: http://127.0.0.1:8200
                     Cgo: disabled
         Cluster Address: https://127.0.0.1:8201
   Environment Variables: BAO_ADDR, HOME, HOSTNAME, PATH
              Go Version: go1.26.5
              Listener 1: tcp (addr: "0.0.0.0:8200", cluster address: "0.0.0.0:8201", max_request_duration: "1m30s", max_request_json_memory: "34078720", max_request_json_strings: "1000", max_request_size: "33554432", tls: "disabled")
               Log Level: trace
           Recovery Mode: false
                 Storage: raft (HA available)
                 Version: OpenBao v2.6.1, committed 2026-07-22T14:22:20Z
             Version Sha: ba7ad8861d0578cd4da4f7b9e5a6756d30484f8f

==> OpenBao server started!
2026-08-07T15:46:21.089Z [DEBUG] would have sent systemd notification (systemd not present): notification=READY=1
2026-08-07T15:46:22.253Z [DEBUG] core: unseal key supplied: migrate=false
2026-08-07T15:46:22.253Z [DEBUG] seals: cannot unseal namespace, not enough keys: keys=1 threshold=2 nonce=fdd8d9d3-33e5-58f7-6c08-4571f2a248e3
2026-08-07T15:46:22.401Z [DEBUG] core: unseal key supplied: migrate=false
2026-08-07T15:46:22.401Z [DEBUG] core: starting cluster listeners
2026-08-07T15:46:22.401Z [INFO]  core.cluster-listener.tcp: starting listener: listener_address=0.0.0.0:8201
2026-08-07T15:46:22.401Z [INFO]  core.cluster-listener: serving cluster requests: cluster_listen_address=[::]:8201
2026-08-07T15:46:22.402Z [TRACE] storage.raft: setting up raft cluster
2026-08-07T15:46:22.402Z [TRACE] storage.raft: applying raft config: inputs="map[node_id:lab path:/openbao/data]"
2026-08-07T15:46:22.402Z [TRACE] storage.raft: using larger timeouts for raft at startup: initial_election_timeout=15s initial_heartbeat_timeout=15s normal_election_timeout=5s normal_heartbeat_timeout=5s
2026-08-07T15:46:22.402Z [INFO]  storage.raft: creating Raft: config="&raft.Config{ProtocolVersion:3, HeartbeatTimeout:15000000000, ElectionTimeout:15000000000, CommitTimeout:50000000, MaxAppendEntries:64, BatchApplyCh:true, ShutdownOnRemove:true, TrailingLogs:0x2800, SnapshotInterval:120000000000, SnapshotThreshold:0x2000, LeaderLeaseTimeout:2500000000, LocalID:\"lab\", NotifyCh:(chan<- bool)(0x1f399bbd2100), LogOutput:io.Writer(nil), LogLevel:\"DEBUG\", Logger:(*hclog.interceptLogger)(0x1f399b5da060), NoSnapshotRestoreOnStart:true, PreVoteDisabled:false, NoLegacyTelemetry:false, skipStartup:false}"
2026-08-07T15:46:22.405Z [INFO]  storage.raft: initial configuration: index=1 servers="[{Suffrage:Voter ID:lab Address:127.0.0.1:8201}]"
2026-08-07T15:46:22.405Z [TRACE] storage.raft: reloaded raft config to set lower timeouts: config="raft.ReloadableConfig{TrailingLogs:0x2800, SnapshotInterval:120000000000, SnapshotThreshold:0x2000, HeartbeatTimeout:5000000000, ElectionTimeout:5000000000}"
2026-08-07T15:46:22.405Z [TRACE] storage.raft: finished setting up raft cluster
2026-08-07T15:46:22.405Z [INFO]  core: vault is unsealed
2026-08-07T15:46:22.405Z [INFO]  storage.raft: entering follower state: follower="Node at 127.0.0.1:8201 [Follower]" leader-address= leader-id=
2026-08-07T15:46:22.405Z [WARN]  storage.raft: heartbeat timeout reached, starting election: last-leader-addr= last-leader-id=
2026-08-07T15:46:22.405Z [INFO]  storage.raft: entering candidate state: node="Node at 127.0.0.1:8201 [Candidate]" term=5
2026-08-07T15:46:22.405Z [DEBUG] storage.raft: pre-voting for self: term=5 id=lab
2026-08-07T15:46:22.405Z [DEBUG] storage.raft: calculated votes needed: needed=1 term=5
2026-08-07T15:46:22.405Z [DEBUG] storage.raft: pre-vote received: from=lab term=5 tally=0
2026-08-07T15:46:22.405Z [DEBUG] storage.raft: pre-vote granted: from=lab term=5 tally=1
2026-08-07T15:46:22.405Z [INFO]  storage.raft: pre-vote successful, starting election: term=5 tally=1 refused=0 votesNeeded=1
2026-08-07T15:46:22.405Z [INFO]  core: entering standby mode
2026-08-07T15:46:22.405Z [INFO]  core: enabling horizontal scalability (reads)
2026-08-07T15:46:22.405Z [INFO]  core: pre-seal teardown starting
2026-08-07T15:46:22.405Z [INFO]  core: stopping raft active node
2026-08-07T15:46:22.405Z [DEBUG] core.invalidate-dispatch: stop triggered
2026-08-07T15:46:22.405Z [DEBUG] core.invalidate-dispatch: finished stopping
2026-08-07T15:46:22.406Z [INFO]  core: pre-seal teardown complete
2026-08-07T15:46:22.406Z [INFO]  core: post-unseal setup starting
2026-08-07T15:46:22.406Z [DEBUG] core: read-only unseal starting
2026-08-07T15:46:22.406Z [INFO]  core: upgrading plugin information: plugins=[]
2026-08-07T15:46:22.406Z [INFO]  core: successfully setup plugin catalog: plugin-directory=/openbao/plugins
2026-08-07T15:46:22.406Z [TRACE] namespace: created dispatcher: name=namespace-deletion-dispatcher num_workers=10
2026-08-07T15:46:22.406Z [TRACE] namespace: initialized dispatcher: num_workers=10
2026-08-07T15:46:22.406Z [TRACE] namespace: created job manager: name=namespace-deletion pool_size=10
2026-08-07T15:46:22.406Z [TRACE] namespace: starting job manager: name=namespace-deletion
2026-08-07T15:46:22.406Z [TRACE] namespace: starting dispatcher
2026-08-07T15:46:22.406Z [INFO]  core: migrating legacy mount table to transactional layout
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=system version="v2.6.1+builtin.bao" path=sys/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=identity version="v2.6.1+builtin.bao" path=identity/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=kv version="v2.6.1+builtin.bao" path=secret/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [ERROR] core: failed to create mount entry: path=agent-registry/ error="plugin not found in the catalog: agent_registry"
2026-08-07T15:46:22.407Z [WARN]  core: skipping plugin-based mount entry: path=agent-registry/
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=agent_registry version="v2.0.3+builtin.vault" path=agent-registry/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [ERROR] core: failed to create mount entry: path=aws/ error="plugin not found in the catalog: aws"
2026-08-07T15:46:22.407Z [WARN]  core: skipping plugin-based mount entry: path=aws/
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=aws version="v2.0.3+builtin.vault" path=aws/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [INFO]  core: successfully mounted: type=cubbyhole version="v2.6.1+builtin.bao" path=cubbyhole/ namespace="ID: root. Path: "
2026-08-07T15:46:22.407Z [TRACE] policy: acquiring lock for: namespace=00000000-0000-0000-0000-000000000000
2026-08-07T15:46:22.407Z [TRACE] policy: acquiring lock for: namespace=00000000-0000-0000-0000-000000000000
2026-08-07T15:46:22.408Z [INFO]  core: migrating legacy auth table to transactional layout
2026-08-07T15:46:22.408Z [TRACE] token: no token generation counter found in storage
2026-08-07T15:46:22.408Z [INFO]  core: successfully mounted: type=token version="v2.6.1+builtin.bao" path=token/ namespace="ID: root. Path: "
2026-08-07T15:46:22.408Z [INFO]  rollback: Starting the rollback manager with 256 workers
2026-08-07T15:46:22.408Z [TRACE] rollback: created dispatcher: name=rollback-dispatcher num_workers=256
2026-08-07T15:46:22.408Z [TRACE] rollback: initialized dispatcher: num_workers=256
2026-08-07T15:46:22.408Z [TRACE] rollback: created job manager: name=rollback pool_size=256
2026-08-07T15:46:22.408Z [TRACE] rollback: starting job manager: name=rollback
2026-08-07T15:46:22.408Z [TRACE] rollback: starting dispatcher
2026-08-07T15:46:22.408Z [TRACE] expiration.job-manager: created dispatcher: name=expire-dispatcher num_workers=200
2026-08-07T15:46:22.408Z [TRACE] expiration.job-manager: initialized dispatcher: num_workers=200
2026-08-07T15:46:22.408Z [TRACE] expiration.job-manager: created job manager: name=expire pool_size=200
2026-08-07T15:46:22.408Z [TRACE] expiration.job-manager: starting job manager: name=expire
2026-08-07T15:46:22.408Z [TRACE] expiration.job-manager: starting dispatcher
2026-08-07T15:46:22.408Z [INFO]  rollback: starting rollback manager
2026-08-07T15:46:22.408Z [DEBUG] storage.raft: voting for self: term=5 id=lab
2026-08-07T15:46:22.409Z [INFO]  core: skipping lease restoration - standby
2026-08-07T15:46:22.409Z [DEBUG] core: loading entities: namespace=""
2026-08-07T15:46:22.409Z [DEBUG] core: entities collected: num_existing=0
2026-08-07T15:46:22.409Z [INFO]  core: entities restored
2026-08-07T15:46:22.409Z [DEBUG] core: identity loading groups: namespace=""
2026-08-07T15:46:22.409Z [DEBUG] core: groups collected: num_existing=0
2026-08-07T15:46:22.409Z [INFO]  core: groups restored
2026-08-07T15:46:22.409Z [DEBUG] core: identity loading OIDC clients: namespace=""
2026-08-07T15:46:22.409Z [TRACE] mfa: loading login MFA configurations
2026-08-07T15:46:22.409Z [TRACE] mfa: methods collected: num_existing=0
2026-08-07T15:46:22.409Z [TRACE] mfa: configurations restored: namespace="" prefix=login-mfa/method/
2026-08-07T15:46:22.409Z [TRACE] mfa: loading login MFA enforcement configurations
2026-08-07T15:46:22.409Z [TRACE] mfa: enforcements configs collected: num_existing=0
2026-08-07T15:46:22.409Z [TRACE] mfa: enforcement configurations restored: namespace="" prefix=login-mfa/enforcement/
2026-08-07T15:46:22.409Z [TRACE] core.invalidate-dispatch: created dispatcher: name=invalidate-dispatch-dispatcher num_workers=128
2026-08-07T15:46:22.409Z [TRACE] core.invalidate-dispatch: initialized dispatcher: num_workers=128
2026-08-07T15:46:22.409Z [TRACE] core.invalidate-dispatch: created job manager: name=invalidate-dispatch pool_size=128
2026-08-07T15:46:22.409Z [TRACE] core.invalidate-dispatch: starting job manager: name=invalidate-dispatch
2026-08-07T15:46:22.409Z [TRACE] core.invalidate-dispatch: starting dispatcher
2026-08-07T15:46:22.410Z [ERROR] core: skipping initialization for nil backend: path=aws/ type=aws version="v2.0.3+builtin.vault" path=aws/
2026-08-07T15:46:22.410Z [ERROR] core: skipping initialization for nil backend: path=agent-registry/ type=agent_registry version="v2.0.3+builtin.vault" path=agent-registry/
2026-08-07T15:46:22.410Z [INFO]  core: post-unseal setup complete
2026-08-07T15:46:22.410Z [TRACE] core: found new active node information, refreshing
2026-08-07T15:46:22.414Z [DEBUG] storage.raft: vote granted: from=lab term=5 tally=1
2026-08-07T15:46:22.414Z [INFO]  storage.raft: election won: term=5 tally=1
2026-08-07T15:46:22.414Z [INFO]  storage.raft: entering leader state: leader="Node at 127.0.0.1:8201 [Leader]"
2026-08-07T15:46:22.427Z [INFO]  core: acquired lock, enabling active operation
2026-08-07T15:46:22.427Z [TRACE] core.invalidate-dispatch: processing invalidation: key=core/lock
2026-08-07T15:46:22.427Z [DEBUG] core.invalidate-dispatch: core context canceled, skipping job: key=core/lock
2026-08-07T15:46:22.427Z [TRACE] core.invalidate-dispatch: concluding processing of invalidation: key=core/lock
2026-08-07T15:46:22.427Z [DEBUG] core.invalidate-dispatch: core context canceled, skipping pending queue processing
2026-08-07T15:46:22.428Z [DEBUG] core: generating cluster private key
2026-08-07T15:46:22.434Z [DEBUG] core: generating local cluster certificate: host=fw-7466014b-7e6f-3b0b-ad24-e0103271418c
2026-08-07T15:46:22.449Z [INFO]  core: pre-seal teardown starting
2026-08-07T15:46:22.449Z [INFO]  core: stopping raft active node
2026-08-07T15:46:22.449Z [TRACE] namespace: terminating job manager...
2026-08-07T15:46:22.449Z [TRACE] namespace: terminating dispatcher
2026-08-07T15:46:22.449Z [DEBUG] core.invalidate-dispatch: stop triggered
2026-08-07T15:46:22.449Z [TRACE] core.invalidate-dispatch: terminating job manager...
2026-08-07T15:46:22.449Z [TRACE] core.invalidate-dispatch: terminating dispatcher
2026-08-07T15:46:22.449Z [DEBUG] core.invalidate-dispatch: finished stopping
2026-08-07T15:46:22.449Z [DEBUG] expiration: stop triggered
2026-08-07T15:46:22.449Z [TRACE] expiration.job-manager: terminating job manager...
2026-08-07T15:46:22.449Z [TRACE] expiration.job-manager: terminating dispatcher
2026-08-07T15:46:22.449Z [DEBUG] expiration: finished stopping
2026-08-07T15:46:22.449Z [INFO]  rollback: stopping rollback manager
2026-08-07T15:46:22.449Z [TRACE] rollback: terminating job manager...
2026-08-07T15:46:22.449Z [TRACE] rollback: terminating dispatcher
2026-08-07T15:46:22.450Z [INFO]  core: pre-seal teardown complete
2026-08-07T15:46:22.450Z [INFO]  core: post-unseal setup starting
2026-08-07T15:46:22.450Z [DEBUG] core: standard unseal starting
2026-08-07T15:46:22.450Z [DEBUG] core: clearing forwarding clients
2026-08-07T15:46:22.450Z [DEBUG] core: done clearing forwarding clients
2026-08-07T15:46:22.450Z [INFO]  core: loaded wrapping token key
2026-08-07T15:46:22.450Z [INFO]  core: upgrading plugin information: plugins=[]
2026-08-07T15:46:22.450Z [INFO]  core: successfully setup plugin catalog: plugin-directory=/openbao/plugins
2026-08-07T15:46:22.450Z [TRACE] namespace: created dispatcher: name=namespace-deletion-dispatcher num_workers=10
2026-08-07T15:46:22.450Z [TRACE] namespace: initialized dispatcher: num_workers=10
2026-08-07T15:46:22.450Z [TRACE] namespace: created job manager: name=namespace-deletion pool_size=10
2026-08-07T15:46:22.450Z [TRACE] namespace: starting job manager: name=namespace-deletion
2026-08-07T15:46:22.450Z [TRACE] namespace: starting dispatcher
2026-08-07T15:46:22.451Z [INFO]  core: migrating legacy mount table to transactional layout
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=system version="v2.6.1+builtin.bao" path=sys/ namespace="ID: root. Path: "
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=identity version="v2.6.1+builtin.bao" path=identity/ namespace="ID: root. Path: "
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=kv version="v2.6.1+builtin.bao" path=secret/ namespace="ID: root. Path: "
2026-08-07T15:46:22.465Z [ERROR] core: failed to create mount entry: path=agent-registry/ error="plugin not found in the catalog: agent_registry"
2026-08-07T15:46:22.465Z [WARN]  core: skipping plugin-based mount entry: path=agent-registry/
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=agent_registry version="v2.0.3+builtin.vault" path=agent-registry/ namespace="ID: root. Path: "
2026-08-07T15:46:22.465Z [ERROR] core: failed to create mount entry: path=aws/ error="plugin not found in the catalog: aws"
2026-08-07T15:46:22.465Z [WARN]  core: skipping plugin-based mount entry: path=aws/
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=aws version="v2.0.3+builtin.vault" path=aws/ namespace="ID: root. Path: "
2026-08-07T15:46:22.465Z [INFO]  core: successfully mounted: type=cubbyhole version="v2.6.1+builtin.bao" path=cubbyhole/ namespace="ID: root. Path: "
2026-08-07T15:46:22.466Z [TRACE] policy: acquiring lock for: namespace=00000000-0000-0000-0000-000000000000
2026-08-07T15:46:22.466Z [TRACE] policy: acquiring lock for: namespace=00000000-0000-0000-0000-000000000000
2026-08-07T15:46:22.466Z [INFO]  core: migrating legacy auth table to transactional layout
2026-08-07T15:46:22.478Z [TRACE] token: no token generation counter found in storage
2026-08-07T15:46:22.478Z [INFO]  core: successfully mounted: type=token version="v2.6.1+builtin.bao" path=token/ namespace="ID: root. Path: "
2026-08-07T15:46:22.478Z [INFO]  rollback: Starting the rollback manager with 256 workers
2026-08-07T15:46:22.478Z [TRACE] rollback: created dispatcher: name=rollback-dispatcher num_workers=256
2026-08-07T15:46:22.478Z [TRACE] rollback: initialized dispatcher: num_workers=256
2026-08-07T15:46:22.478Z [TRACE] rollback: created job manager: name=rollback pool_size=256
2026-08-07T15:46:22.478Z [TRACE] rollback: starting job manager: name=rollback
2026-08-07T15:46:22.478Z [TRACE] rollback: starting dispatcher
2026-08-07T15:46:22.478Z [TRACE] expiration.job-manager: created dispatcher: name=expire-dispatcher num_workers=200
2026-08-07T15:46:22.478Z [TRACE] expiration.job-manager: initialized dispatcher: num_workers=200
2026-08-07T15:46:22.478Z [TRACE] expiration.job-manager: created job manager: name=expire pool_size=200
2026-08-07T15:46:22.478Z [TRACE] expiration.job-manager: starting job manager: name=expire
2026-08-07T15:46:22.478Z [TRACE] expiration.job-manager: starting dispatcher
2026-08-07T15:46:22.478Z [INFO]  core: restoring leases
2026-08-07T15:46:22.479Z [DEBUG] core: loading entities: namespace=""
2026-08-07T15:46:22.479Z [DEBUG] core: entities collected: num_existing=0
2026-08-07T15:46:22.479Z [INFO]  rollback: starting rollback manager
2026-08-07T15:46:22.479Z [DEBUG] expiration: collecting leases
2026-08-07T15:46:22.479Z [DEBUG] expiration: leases collected: num_existing=0 namespaces=1
2026-08-07T15:46:22.479Z [INFO]  expiration: lease restore complete
2026-08-07T15:46:22.479Z [INFO]  core: entities restored
2026-08-07T15:46:22.479Z [DEBUG] core: identity loading groups: namespace=""
2026-08-07T15:46:22.479Z [DEBUG] core: groups collected: num_existing=0
2026-08-07T15:46:22.479Z [INFO]  core: groups restored
2026-08-07T15:46:22.479Z [DEBUG] core: identity loading OIDC clients: namespace=""
2026-08-07T15:46:22.479Z [TRACE] mfa: loading login MFA configurations
2026-08-07T15:46:22.479Z [TRACE] mfa: methods collected: num_existing=0
2026-08-07T15:46:22.479Z [TRACE] mfa: configurations restored: namespace="" prefix=login-mfa/method/
2026-08-07T15:46:22.479Z [TRACE] mfa: loading login MFA enforcement configurations
2026-08-07T15:46:22.479Z [TRACE] mfa: enforcements configs collected: num_existing=0
2026-08-07T15:46:22.479Z [TRACE] mfa: enforcement configurations restored: namespace="" prefix=login-mfa/enforcement/
2026-08-07T15:46:22.479Z [INFO]  core: starting raft active node
2026-08-07T15:46:22.479Z [INFO]  storage.raft: starting autopilot: config="&{false 0 10s 24h0m0s 1000 0 10s}" reconcile_interval=0s
2026-08-07T15:46:22.479Z [DEBUG] storage.raft.autopilot: autopilot is now running
2026-08-07T15:46:22.479Z [DEBUG] storage.raft.autopilot: state update routine is now running
2026-08-07T15:46:22.480Z [DEBUG] core: request forwarding setup function
2026-08-07T15:46:22.480Z [DEBUG] core: clearing forwarding clients
2026-08-07T15:46:22.480Z [DEBUG] core: done clearing forwarding clients
2026-08-07T15:46:22.480Z [DEBUG] core: leaving request forwarding setup function
2026-08-07T15:46:22.480Z [INFO]  core: usage gauge collection is disabled
2026-08-07T15:46:22.491Z [INFO]  core: Recorded vault version: vault version=2.6.1 upgrade time="2026-08-07 15:46:22.480108204 +0000 UTC" commit date=2026-07-22T14:22:20Z
2026-08-07T15:46:22.491Z [ERROR] core: skipping initialization for nil backend: path=agent-registry/ type=agent_registry version="v2.0.3+builtin.vault" path=agent-registry/
2026-08-07T15:46:22.491Z [ERROR] core: skipping initialization for nil backend: path=aws/ type=aws version="v2.0.3+builtin.vault" path=aws/
2026-08-07T15:46:22.491Z [INFO]  core: post-unseal setup complete
2026-08-07T15:46:22.559Z [DEBUG] core: attempting to load backend plugin: name=aws
2026-08-07T15:46:22.559Z [DEBUG] core: spawning a new plugin process: plugin_name=aws id=NqdWMe0we9
2026-08-07T15:46:22.603Z [DEBUG] core: successfully dispensed v5 backend plugin: name=aws
2026-08-07T15:46:22.612Z [DEBUG] core: removed plugin client connection: id=NqdWMe0we9
2026-08-07T15:46:22.612Z [DEBUG] core: killed external multiplexed plugin process: path=/openbao/plugins/openbao-plugin-secrets-aws_linux_amd64_v1 pid=80
2026-08-07T15:46:24.975Z [DEBUG] core: spawning a new plugin process: plugin_name=aws id=rHunu1zGK2
2026-08-07T15:46:24.990Z [DEBUG] secrets.aws.aws_ed580048.aws: configuring client automatic mTLS
2026-08-07T15:46:24.994Z [DEBUG] secrets.aws.aws_ed580048.aws: starting plugin: path=/openbao/plugins/openbao-plugin-secrets-aws_linux_amd64_v1 args=["/openbao/plugins/openbao-plugin-secrets-aws_linux_amd64_v1"]
2026-08-07T15:46:24.994Z [DEBUG] secrets.aws.aws_ed580048.aws: plugin started: path=/openbao/plugins/openbao-plugin-secrets-aws_linux_amd64_v1 pid=126
2026-08-07T15:46:24.994Z [DEBUG] secrets.aws.aws_ed580048.aws: waiting for RPC address: plugin=/openbao/plugins/openbao-plugin-secrets-aws_linux_amd64_v1
2026-08-07T15:46:25.000Z [INFO]  secrets.aws.aws_ed580048.aws.openbao-plugin-secrets-aws_linux_amd64_v1: configuring server automatic mTLS: timestamp=2026-08-07T15:46:25.000Z
2026-08-07T15:46:25.010Z [DEBUG] secrets.aws.aws_ed580048.aws: using plugin: version=5
2026-08-07T15:46:25.019Z [TRACE] secrets.aws.aws_ed580048.aws.stdio: waiting for stdio data
2026-08-07T15:46:25.020Z [TRACE] secrets.aws.aws_ed580048.aws: setup: transport=gRPC status=started
2026-08-07T15:46:25.029Z [TRACE] secrets.aws.aws_ed580048.aws: setup: transport=gRPC status=finished err=<nil> took=9.015025ms
2026-08-07T15:46:25.039Z [TRACE] secrets.aws.aws_ed580048.aws: initialize: transport=gRPC status=started
2026-08-07T15:46:25.039Z [TRACE] secrets.aws.aws_ed580048.aws: initialize: transport=gRPC status=finished err=<nil> took="650.516µs"
2026-08-07T15:46:25.039Z [TRACE] secrets.aws.aws_ed580048.aws: special paths: transport=gRPC status=started
2026-08-07T15:46:25.040Z [TRACE] secrets.aws.aws_ed580048.aws: special paths: transport=gRPC status=finished took="315.885µs"
2026-08-07T15:46:25.040Z [INFO]  core: successfully reloaded plugin: plugin=aws_ed580048 path=aws/ version=v0.3.1
2026-08-07T15:46:25.195Z [TRACE] secrets.aws.aws_ed580048.aws: handle existence check: transport=gRPC path=config/root status=started
2026-08-07T15:46:25.197Z [TRACE] secrets.aws.aws_ed580048.aws: handle existence check: transport=gRPC path=config/root status=finished err=<nil> took=1.424662ms
2026-08-07T15:46:25.197Z [TRACE] secrets.aws.aws_ed580048.aws: handle request: transport=gRPC path=config/root status=started
2026-08-07T15:46:25.198Z [TRACE] secrets.aws.aws_ed580048.aws: handle request: transport=gRPC path=config/root status=finished err="cannot write to storage during setup" took=1.460691ms
2026-08-07T15:46:30.351Z [TRACE] secrets.aws.aws_ed580048.aws: handle existence check: transport=gRPC path=config/root status=started
2026-08-07T15:46:30.351Z [TRACE] secrets.aws.aws_ed580048.aws: handle existence check: transport=gRPC path=config/root status=finished err=<nil> took="834.578µs"
2026-08-07T15:46:30.352Z [TRACE] secrets.aws.aws_ed580048.aws: handle request: transport=gRPC path=config/root status=started
2026-08-07T15:46:30.353Z [TRACE] secrets.aws.aws_ed580048.aws: handle request: transport=gRPC path=config/root status=finished err="cannot write to storage during setup" took=1.601255ms
```

Reviewer: Right, I think that just loops back to the problem I'm trying to solve in https://github.com/openbao/openbao/pull/3486.

Works for me if we just document that a restart is necessary (IIUC that worked for you?)

Author: OK, works for me.

Vault 2.0 and newer create an `agent-registry/` mount on their own, so a store coming

> from a current Vault has at least one such entry even if you never enabled one.

Reviewer: ```suggestion
Vault 2.0 and later enable an `agent-registry/` mount by default, so a
migration from a current Vault yields at least one such entry.
```
